logoapp/js/common_data.js

```diff
@@ -0,0 +1,175 @@
+//const
+var DOT_NUM = 19;
+var LINE_NUM =42;
+var ORIGINAL_CANVAS_SIZE = 500;
+var MAINCANVAS_WIDTH = 450;
+var MAINCANVAS_HEIGHT	 = 450;
+var MAIN_SCALE = MAINCANVAS_WIDTH/ORIGINAL_CANVAS_SIZE;
+var PREVIEWCANVAS_WIDTH = 48;
+var PREVIEWCANVAS_HEIGHT = 48;
+var PREVIEW_SCALE = PREVIEWCANVAS_WIDTH/ORIGINAL_CANVAS_SIZE;
+var LINE_LENGTH = ORIGINAL_CANVAS_SIZE*0.1777;
+var DOT_RADIUS = LINE_LENGTH*0.125;
+var LINE_DRAW_SCALE = 0.72;
+var WHITE = '#fff';
+var KIDS_GREEN = '#9c3';
+var MIRAI_ORANGE = '#f90';
+var BACKGROUND_COLOR = WHITE;
+var DRAW_COLOR = MIRAI_ORANGE;
+
+Vector2 = function(x, y) {
+	this.x = x;
+	this.y = y;
+};
+Vector2.prototype.dist = function() {
+	return Math.sqrt((this.x*this.x)+(this.y*this.y));
+};
+
+function GetCrossProductByPoint(P, A, B) {
+	var PA = new Vector2(P.x-A.x, P.y-A.y);
+	var BA = new Vector2(B.x-A.x, B.y-A.y);
+	var crsP = (PA.x*BA.y)-(BA.x*PA.y);
+	return crsP;
+};
+function GetCosByVector(O, A, B) {
+	var a = new Vector2(A.x-O.x, A.y-O.y);
+	var b = new Vector2(B.x-O.x, B.y-O.y);
+	var dot = (a.x*b.x)+(a.y*b.y)
+	var cos = dot/ (a.dist()*b.dist());
+	return cos;
+};
+function isAcuteAngleTriangle(P, A, B) {
+	var cos1 = GetCosByVector(A, P, B);
+	var cos2 = GetCosByVector(B, P, A);
+	if(cos1*cos2<0) return false;
+	else return true;
+};
+function getDotPos(num, interval) {
+	var pos = new Vector2(0,0);
+	//var angle = [-90, -30, 30, 90, 150, 210];
+	if(num==0) {
+		pos.x=0;
+		pos.y=0;
+	} else if(num>0 && num<7) {
+		pos.x = Math.cos(((60*(num-1))-90) * Math.PI / 180) * interval;
+		pos.y = Math.sin(((60*(num-1))-90) * Math.PI / 180) * interval;
+	} else if(num>6) {
+		if(num%2==1) {
+			pos.x = Math.cos(((60*(num-7)/2)-90) * Math.PI / 180) * interval;
+			pos.y = Math.sin(((60*(num-7)/2)-90) * Math.PI / 180) * interval;
+			pos.x += Math.cos(((60*(num-7)/2)-90) * Math.PI / 180) * interval;
+			pos.y += Math.sin(((60*(num-7)/2)-90) * Math.PI / 180) * interval;
+		} else if(num%2==0) {

```

Comment:
__[jshint]__
*  Use '===' to compare with '0'.


<!-- Comment by [SideCI](https://sideci.com) -->